File: es-0/person.py
"""
modulo che contiene la classe persona
"""
class Person:
	''' CLASSE PERSONA '''
	# costruttore della classe
	def __init__(self, firstname, lastname, age = 18):
		self.__firstname = firstname
		self.__lastname = lastname
		self.__age = age

	# metodo toString()
	def __str__(self):
		return f'Sono {self.__firstname} {self.__lastname}, ho {self.__age} anni'

	# getters
	def get_firstname(self):
		return self.__firstname

	def get_lastname(self):
		return self.__lastname

	def get_age(self):
		return self.__age

	# getters
	def set_firstname(self, firstname):
		if len(strip(firstname)) >= 3:
			self.__firstname = strip(firstname)

	def set_lastname(self, lastname):
		if len(strip(lastname)) >= 3:
			self.__lastname = strip(lastname)

	def set_age(self, age):
		if age > 0:
			self.__age = age

'''
il main verrà richiamato solo se questo file verrà lanciato da linea di comando.
nel caso venga lanciato come uno script o un modulo non verrà invocato
'''
# main()
if __name__ == '__main__':
	teacher = Person('Biagio Rosario', 'Greco', 41)
	print(teacher)

File: info-5x/es-20/spring/1stwebapp/src/main/java/model/Person.java
package model;
public class Person {
	private long id;
	private String firstname, lastname;
	private int age;
	public Person(long id, String firstname, String lastname, int age) {
		this.id = id > 0 ? id : 0;
		this.firstname = !firstname.isBlank() ? firstname.strip() : "(no firstname)";
		this.lastname = !lastname.isBlank() ? lastname.strip() : "(no lastname)";
		this.age = age > 0 ? age : 0;
	}
	public long getId() {
		return this.id;
	}
	public String getFirstname() {
		return this.firstname;
	}
	public String getLastname() {
		return this.lastname;
	}
	public int getAge() {
		return this.age;
	}
	public void setFirstname(String firstname) {
		if(!firstname.isBlank())
			this.firstname = firstname.strip();
	}
	public void setLastname(String lastname) {
		if(!lastname.isBlank())
			this.lastname = lastname.strip();
	}
	public void setAge(int age) {
		if(age > 0)
			this.age = age;
	}
	public void setPerson(Person p) {
		setFirstname(p.firstname);
		setLastname(p.lastname);
		setAge(p.age);
	}
	@Override
	public String toString() {
		return "[%d] Hi, I'm %s %s. I'm %d!..".formatted(id, firstname, lastname, age);
	}	
}